lib/fileSystem.js

```diff
@@ -0,0 +1,115 @@
+var q = require('q');
+
+/**
+ * @param {object} e error object
+ */
+function _fsErrorHandler(e) {
+  var msg = '';
+
+  switch (e.code) {
+  case FileError.QUOTA_EXCEEDED_ERR:
+    msg = 'QUOTA_EXCEEDED_ERR';
+    break;
+  case FileError.NOT_FOUND_ERR:
+    msg = 'NOT_FOUND_ERR';
+    break;
+  case FileError.SECURITY_ERR:
+    msg = 'SECURITY_ERR';
+    break;
+  case FileError.INVALID_MODIFICATION_ERR:
+    msg = 'INVALID_MODIFICATION_ERR';
+    break;
+  case FileError.INVALID_STATE_ERR:
+    msg = 'INVALID_STATE_ERR';
+    break;
+  default:
+    msg = 'Unknown Error';
+    break;
+  }
+
+  throw new Error('Error: ' + msg);
+}
+
+function _requestFileSystem() {
+  var d = q.defer();
+  navigator.webkitPersistentStorage.requestQuota(1024 * 1024, function(grantedBytes) {
+    window.webkitRequestFileSystem(PERSISTENT, grantedBytes, function(fs) {
+      d.resolve(fs);
+    }, _fsErrorHandler);
+  }, function(e) {
+    d.reject(e);
+  });
+  return d.promise;
+}
+
+var fileSystem = {};
+
+fileSystem.init = function() {
+  return _requestFileSystem().then(function(fs) {
+    fileSystem.ready = true;
+    fileSystem.fs = fs;
+  }, function(err) {
+    fileSystem.ready = false;
+    throw new Error('Cannot reserve space on fileSystem' + err);
+  });
+};
+
+/**
+ * @param {string} name fileName/key
+ * @param {object} data queue data
+ */
+fileSystem.writeFile = function(name, contentType, data) {
+  var deferredWrite = q.defer();
+  var fileData = JSON.stringify(data);
+
+  if (fileSystem.ready) {
```

Comment:
There is no else here, the promise will never resolve.

You might consider adding a promise to the `fileSystem.init` function to return a promise instead of setting a ready flag.

In that case, anything using that promise would immediately reject with a useful error message.